packages/flutter_tools/lib/src/windows/topaz_ofd_validator.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+const String kCoreProcessPattern = r'Topaz OFD\\Warsaw\\core.exe';
+
+class TopazOfdValidator extends DoctorValidator {
+  const TopazOfdValidator({required this.processLister}) : super('Topaz OFD');
+
+  final ProcessLister processLister;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final String tasks = await processLister.getProcessesWithPath('core');
+    final RegExp pattern = RegExp(kCoreProcessPattern, multiLine: true, caseSensitive: false);
+    final bool matches = pattern.hasMatch(tasks);
+    if (matches) {
+      return const ValidationResult(
+        ValidationType.partial,
+        <ValidationMessage>[
+          ValidationMessage.hint('The Topaz OFD Security Module process has been found running. If you are unable to build, you will need to disable it.'),
+        ],
+        statusInfo: 'Topaz OFD may be running');
+    } else {
+      return const ValidationResult(
+        ValidationType.success,
+        <ValidationMessage>[]);
+    }
+  }
+}
+
+class ProcessLister {
+  Future<String> getProcessesWithPath(String? filter) async {
+    final String argument = filter == null ? 'Get-Process $filter | Format-List Path' : 'Get-Process | Format-List Path';
+    final ProcessResult taskResult = await Process.run('powershell', <String>['-command', argument]);
```

Comment:
Could the subprocess fail with an error code? How should that be handled?

Reply:
What are your thoughts on returning `partial` with a message indicating that the check failed when the exit code is non-zero?

Reply:
Looking at some of the other examples, I think `missing` might make more sense? Wondering if @christopherfujino has an opinion.

Reply:
Yeah, I think missing, because we'd want to investigate further and see if this approach is not robust for all windows setups.

Reply:
So `missing` when the exit code is nonzero, but is `partial` still desired for when Topaz is found?

Reply:
I think missing in both cases. In my mind, partial is just if a validator has sub-validators, with mixed results.